### DISCUSSION TITLE
how to view entire decision region

### DISCUSSION THREAD

**Discussion by ap266** [rasbt/mlxtend#968] [Q&A] [answered]
Hello, using plot_decision_region, how can one view what the decision region looks like for examples that are outside of the trained distribution. For the XOR function book example, y(i) = 0 if x0(i) * x1(i) < 0 otherwise 1, with training values between (-1, 1), I'd like to view the decision region for test values 10x smaller and larger than (-1, 1). In the provided screenshot of a decision boundary created by 2000 (-1, 1) examples, a model.predict([[-2, .03]]) would result in a guess of 1 rather than 0. The decision boundary extends outward to larger numbers because even tests like (-100, 3) results in 1 rather than 0.  

![screenshot](https://user-images.githubusercontent.com/40244861/187096916-dcce49c2-1c34-4c7f-a28b-e45069943b38.png)

**Comment by rasbt** [comment, marked as answer]
Good question, I think the `zoom_factor` param could help you with this.

**Comment by ap266** [comment]
I was able to use the zoom_factor feature in the latest build of mlxtend. I was using the default mlxtend on Colab and just upgraded to the latest build and that worked!

**Reply by rasbt** [reply]
Awesome, glad to hear!
